src/ImageIO/CGImageAnimation.cs

```diff
@@ -0,0 +1,141 @@
+//
+// ImageIO - CGImageAnimation.cs
+//
+// Authors:
+//	Whitney Schmidt  <whschm@microsoft.com>
+//
+// Copyright 2020, Microsoft Corp.
+//
+
+using System;
+using System.Runtime.InteropServices;
+using CoreFoundation;
+using CoreGraphics;
+using Foundation;
+using ObjCRuntime;
+
+namespace ImageIO
+{
+
+    public partial class CGImageAnimation
+    {
+
+        public delegate void CGImageSourceAnimationBlock (nint index, CGImage image, out bool stop);
+
+        [Introduced (PlatformName.MacOSX, 10, 15, PlatformArchitecture.All)]
+        [Introduced (PlatformName.iOS, 13, 0, PlatformArchitecture.All)]
+        [DllImport (Constants.ImageIOLibrary)]
+        static extern int CGAnimateImageAtURLWithBlock ( /* CFURLRef */ IntPtr url, /* CFDictionaryRef _iio_Nullable */ IntPtr options, /* CGImageSourceAnimationBlock */ ref BlockLiteral block);
+
+        [Introduced (PlatformName.MacOSX, 10, 15, PlatformArchitecture.All)]
+        [Introduced (PlatformName.iOS, 13, 0, PlatformArchitecture.All)]
+        [DllImport (Constants.ImageIOLibrary)]
+        static extern int CGAnimateImageDataWithBlock ( /* CFDataRef _Nonnull */ IntPtr data, /* CFDictionaryRef _Nullable */ IntPtr options, /* CGImageSourceAnimationBlock _Nonnull */ ref BlockLiteral block);
+
+        [Introduced (PlatformName.MacOSX, 10, 15, PlatformArchitecture.All)]
+        [Introduced (PlatformName.iOS, 13, 0, PlatformArchitecture.All)]
+        [BindingImpl (BindingImplOptions.Optimizable)]
+        public int AnimateImage (NSUrl url, NSDictionary options, [BlockProxy (typeof (NIDCGImageSourceAnimationBlock))] CGImageSourceAnimationBlock block)
```

Comment:
This returns an `int` which in reality it is an `OSStatus` but I wonder if we can do even better, there is `CGImageAnimationStatus` inside `CGImageAnimation.h` I think it is better to return `CGImageAnimationStatus` 

@spouliot what do you think?

Reply:
@dalexsoto @spouliot this seems like a good idea. One more question: the enum looks like this:

```
public enum CGImageAnimationStatus
	{
		ParameterError = -22140,
		CorruptInputImage = -22141,
		UnsupportedFormat = -22142,
		IncompleteInputImage = -22143,
		AllocationFailure = -22144,
	}
```

It seems like `None = 0` should be added, thoughts?

Reply:
Good point! but looks at `OSStatus` enum to see how `0` was named (for consistency).

Reply:
Added `Ok = 0` (thanks for the FYI on `enums` @dalexsoto!). 

Looks like we use `Ok` normally in our code: https://github.com/xamarin/xamarin-macios/blob/5eb593f4ba86ef2c3dd9554db1ab5a9cbb9df923/src/VideoToolbox/VTDefs.cs#L17-L18